protobuf-kafkaconnect-converter/src/test/java/com/amazonaws/services/schemaregistry/kafkaconnect/protobuf/ToProtobufTestDataGenerator.java

```diff
@@ -642,4 +642,165 @@ private static Map<String, Schema> getOneofType() {
+                .setField(descriptor.findFieldByName("date"), dateBuilder.build())
+                .setField(descriptor.findFieldByName("time"), todBuilder.build())
+                .setField(descriptor.findFieldByName("timestamp"), timestampBuilder.build())
+                .addRepeatedField(descriptor.findFieldByName("intMap"), intMapBuilder.build())
+                .addRepeatedField(descriptor.findFieldByName("boolMap"), boolMapBuilder.build())
+                .addRepeatedField(descriptor.findFieldByName("boolMap"), boolMapBuilder2.build())
+                .setField(descriptor.findFieldByName("color"), descriptor.findEnumTypeByName("Colors").findValueByName("BLACK"))
+                .setField(descriptor.findFieldByName("progress"), fileDescriptor.findEnumTypeByName("Progress").findValueByName("INPROGRESS"))
+                .setField(descriptor.findFieldByName("id"), 12315)
+                .setField(descriptor.findFieldByName("address"), addressBuilder.build())
+                .setField(descriptor.findFieldByName("customer"), customerBuilder.build())
+                .build();
+    }
+
+    private static Descriptors.FileDescriptor getAllTypesFileDescriptor(String schemaName) {
+        return new ConnectSchemaToProtobufSchemaConverter().convert(getAllTypesSchema(schemaName));
+    }
+
+    public static Struct getAllTypesData(String schemaName) {
+        Schema connectSchema = getAllTypesSchema(schemaName);
+        final Struct connectData = new Struct(connectSchema);
+
+        int dateDefVal = 19071; // equal to 2022/03/20 with reference to the unix epoch
+        int timeDefVal = 7362000; // equal to 2 hours 2 minutes 42 seconds in millisecond
+        long tsDefVal = 1805; // equal to 1 second 805000000 nanoseconds in millisecond
+        java.util.Date date = Date.toLogical(Date.SCHEMA, dateDefVal);
+        java.util.Date time = Time.toLogical(Time.SCHEMA, timeDefVal);
+        java.util.Date timestamp = Timestamp.toLogical(Timestamp.SCHEMA, tsDefVal);
+
+        connectData
+                .put("i32", 32)
+                .put("bool", false)
+                .put("bytes", new byte[] { 1, 5, 6, 7 })
+                .put("str", "Hello world!")
+                .put("strArray", Arrays.asList("foo", "bar", "baz"))
+                .put("boolArray", Arrays.asList(true, false))
+                .put("intArray", new ArrayList<>())
+                .put("date", date)
+                .put("time", time)
+                .put("timestamp", timestamp)
+                .put("intMap", ImmutableMap.of(2, 22))
+                .put("boolMap", ImmutableMap.of("A", true, "B", false))
+                .put("strMap", new HashMap<>())
+                .put("color", "BLACK")
+                .put("progress", "INPROGRESS")
+                .put("order", new Struct(connectSchema.field("order").schema()).put("id", 12315))
+                .put("address", new Struct(connectSchema.field("address").schema()).put("street", "8th").put("zipcode", 98121))
```

Comment:
Can we add an enum and all the fields in the same schema? Comment applicable everywhere.

Reply:
Yes they are included in the AllTypesSchema